File: app/src/main/java/com/boardgamegeek/service/CollectionUploadTask.java
package com.boardgamegeek.service;

import android.support.annotation.NonNull;

import okhttp3.FormBody;
import okhttp3.OkHttpClient;

public abstract class CollectionUploadTask extends CollectionTask {

	public CollectionUploadTask(OkHttpClient client) {
		super(client);
	}

	public abstract String getTimestampColumn();

	public abstract boolean isDirty();

	@NonNull
	protected FormBody.Builder createFormBuilder() {
		return super.createFormBuilder()
			.add("action", "savedata")
			.add("collid", String.valueOf(collectionItem.getCollectionId()));
	}
}

File: app/src/main/java/com/boardgamegeek/service/CollectionUploadTask.kt
package com.boardgamegeek.service

import com.boardgamegeek.entities.CollectionItemForUploadEntity
import okhttp3.FormBody
import okhttp3.OkHttpClient

abstract class CollectionUploadTask(client: OkHttpClient) : CollectionTask(client, CollectionItemForUploadEntity()) {
    abstract val timestampColumn: String

    abstract val isDirty: Boolean

    open fun addCollectionItem(collectionItem: CollectionItemForUploadEntity) {
        this.collectionItem = collectionItem
    }

    @Suppress("SpellCheckingInspection")
    override fun createFormBuilder(): FormBody.Builder {
        return super.createFormBuilder()
                .add("action", "savedata")
                .add("collid", collectionItem.collectionId.toString())
    }
}